### PR TITLE
Add Boot Environment Integration to OPNsense Web Interface

### PR DESCRIPTION
**Pull request by sheridans**
This pull request introduces a new feature to the OPNsense web interface, allowing users to manage FreeBSD boot environments directly within OPNsense. This integration provides an intuitive and seamless way for users to create, manage, and switch between boot environments, enhancing system management and recovery options.

### CODE REVIEW

**Comment by AdSchellevis** [comment]
@sheridans as this is quite some code, we will need time for reviewing and testing. It sounds like a practical addition to OPNsense, probably roadmap 25.1 material.

File: src/opnsense/mvc/app/controllers/OPNsense/BootEnvironments/Api/GeneralController.php [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+                'path' => $params,
+                'rows' => $backResult,
+                'current' => 1,
+                'total' => count($backResult),
+                'rowCount' => 7,
+                'searchPhrase' => ''
+            ];
+        }
+        return ['message' => 'Unable to list boot environments'];
+    }
```
Comment (fichtner) [inline src/opnsense/mvc/app/controllers/OPNsense/BootEnvironments/Api/GeneralController.php:85]:
```suggestion
    }
```

test

Comment (fichtner) [reply src/opnsense/mvc/app/controllers/OPNsense/BootEnvironments/Api/GeneralController.php:85]: @sheridans nope, won't let me add this to the PR. there is a setting somewhere at the right to enable maintainer edits

Comment (sheridans) [reply src/opnsense/mvc/app/controllers/OPNsense/BootEnvironments/Api/GeneralController.php:85]: Screenshot_20240726_125431_Firefox

It's ticked?

Comment (fichtner) [reply src/opnsense/mvc/app/controllers/OPNsense/BootEnvironments/Api/GeneralController.php:85]: Ok, appears to work then. It won’t let me commit empty changes then. Sorry for the noise.

File: src/opnsense/mvc/app/models/OPNsense/BootEnvironments/General.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<model>
+    <mount>//OPNsense/BootEnvironments/general</mount>
+    <description>
+        A mesh VPN that makes it easy to connect your devices, wherever they are.
```
Comment (Monviech) [inline src/opnsense/mvc/app/models/OPNsense/BootEnvironments/General.xml:4]: Probably a typo here?

Comment (sheridans) [reply src/opnsense/mvc/app/models/OPNsense/BootEnvironments/General.xml:4]: Yes, copied that file from the plugin I'm working on for Tailscale and failed to change the description. Will fix.

**Comment by AdSchellevis** [comment]
@sheridans I've created a new branch (be) to work on this, but it does need quite some effort before it will be in an acceptable state for core. 

I don't mind restructuring the code and making this happen somewhere on the road to 25.1, but I do want to make sure I'm not running into obstructions while doing so. My first batch of changes (already showing a bit of the direction) is committed in https://github.com/opnsense/core/commit/6ce7644dd9a225f79ef6b71bf968553aafa12fe5

Thoughts?

**Comment by sheridans** [comment]
I can see where you're going with this, and how the code can be cleaned up. It was my first attempt at writing anything for OPNsense, from your changes I can see the direction. I'll take a look at refactoring, I had actually gotten the model working for validation and things, but this can be done in code without any issue. I'd already addressed most of the issue with validation, checking if they exist etc as you branched it. I'll create a PR into that branch, thanks for pointing in the right direction, exactly what I needed

**Comment by AdSchellevis** [comment]
@sheridans I also don't mind finishing the work needed to pull this trough, just don't want to step on anyones toes.

**Comment by sheridans** [comment]
Heh, there's no toes to be stepped on here. Open source is a team sport. I'm happy  to finish it, or let it be finished.

The goal, once functional, was to introduce boot environments with auto recovery features [without a paid subscription, mentioning no company that may or may not have inspired this :)].

If you're schedule is limited, or struggling time wise, happy to help wrap it up. Otherwise, have at it...

**Comment by AdSchellevis** [comment]
let me take another stab at it then, I'll keep you posted.

**Comment by AdSchellevis** [comment]
superseded by https://github.com/opnsense/core/pull/7749

**Pull request closed without merging**